stream-chat-android-client/src/main/java/io/getstream/chat/android/client/call/DistinctCall.kt

```diff
@@ -105,4 +107,8 @@ internal class DistinctCall<T : Any>(
     private companion object {
         private const val TAG = "Chat:DistinctCall"
     }
+
+    override suspend fun await(): Result<T> = withContext(DispatcherProvider.IO) {
+        execute()
+    }
```

Comment:
Hey @JcMinarro,

Can we turn `await` into the way of using `suspendCancellableCoroutine`:
```suggestion
    override suspend fun await(): Result<T> {
        StreamLog.d(TAG) { "[await] uniqueKey: $uniqueKey" }
        return suspendCancellableCoroutine { continuation ->
            enqueue { result ->
                StreamLog.v(TAG) { "[await] completed($uniqueKey)" }
                continuation.resume(result)
            }
            continuation.invokeOnCancellation {
                StreamLog.v(TAG) { "[await] canceled($uniqueKey)" }
                delegate.get()?.cancel()
            }
        }
    }
```

and replace `execute` with the following snippet:
```kotlin
    override fun execute(): Result<T> = runBlocking {
        await()
    }
```

Reply:
This class has been refactored to follow the same approach than the rest of implementations